### PR TITLE
I18n expansion syntax

### PR DESCRIPTION
**Pull request by vsavkin**
Add support for 

```
{messages.length, plural,
  =0 {zero}
  =1 {one}
  =other {more than 1}
}
```

### CODE REVIEW

**Comment by mhevery** [comment]
@kara can you have a look as well?

File: modules/angular2/test/compiler/html_lexer_spec.ts [hunk 1]
```diff
@@ -576,6 +576,64 @@ export function main() {
...
+              [HtmlTokenType.EXPANSION_CASE_VALUE, '4'],
+              [HtmlTokenType.EXPANSION_CASE_EXP, 'four'],
+              [HtmlTokenType.EXPANSION_FORM_END],
+              [HtmlTokenType.TEXT, "after"],
+              [HtmlTokenType.EOF]
+            ]);
+      });
+
+      it("should parse an expansion forms with elements in it", () => {
+        expect(tokenizeAndHumanizeParts('{one.two, three, =4 {four <b>a</b>}}', true))
```
Comment (mhevery) [inline modules/angular2/test/compiler/html_lexer_spec.ts]: The HTML embedded in the i18n will need to be parsed.

File: modules/angular2/test/compiler/html_parser_spec.ts [hunk 2]
```diff
@@ -240,6 +242,18 @@ export function main() {
         });
       });
 
+      describe("expansion forms", () => {
+        it("html parser should parse out crazy expressions", () => {
```
Comment (mhevery) [inline modules/angular2/test/compiler/html_parser_spec.ts]: could you explain `crazy`?

File: modules/angular2/test/i18n/i18n_html_parser_spec.ts [hunk 3]
```diff
@@ -158,6 +158,40 @@ export function main() {
       expect(res[1].sourceSpan.start.offset).toEqual(10);
     });
 
+    it("should handle the plural special form", () => {
+      let translations: {[key: string]: string} = {};
+      translations[id(new Message('zero', "plural_0", null))] = 'ZERO';
+
+      let res = parse(`{messages.length, plural,=0 {zero}}`, translations);
+
+      expect(humanizeDom(res))
```
Comment (mhevery) [inline modules/angular2/test/i18n/i18n_html_parser_spec.ts]: rename to `humanizeAST`

File: modules/angular2/test/compiler/html_lexer_spec.ts (hunk 1 above)
Comment (vsavkin) [inline modules/angular2/test/compiler/html_lexer_spec.ts]: Fixed it

**Pull request closed without merging**